### PR TITLE
converted src/components/App.js to es6

### PR DESCRIPTION
**Pull request by AnshulMalik**
Associated Issue: #1890

### Summary of Changes

* Migrated `src/components/App.js` to use es6 classes

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const { onConnect, onFirefoxConnect } = require("./utils/client");
 const reducers = require("./reducers");
 const selectors = require("./selectors");
 
-const App = require("./components/App");
+const App = require("./components/App").default;
```
Comment (wldcordeiro) [inline src/main.js]: If you use import/export for the `App` file I think changing this line  to
`import App from './components/App'` isn't bad.

Comment (AnshulMalik) [inline src/main.js]: I thought I'd do that when I migrate main.js file. But essentially the same thing.

Comment (wldcordeiro) [inline src/main.js]: It kind of acts as an indicator that `App` was done. 😄

File: src/components/App.js [hunk 2]
```diff
@@ -26,38 +26,35 @@ const SecondaryPanes = createFactory(require("./SecondaryPanes"));
...
-  },
+class App extends React.Component {
+  state: {
+    horizontal: verticalLayoutBreakpoint.matches
+  }
 
-  displayName: "App",
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (wldcordeiro) [inline src/components/App.js]: Doesn't this duplicate the work above at line 30?

File: src/components/App.js [hunk 3]
```diff
@@ -135,19 +132,29 @@ const App = React.createClass({
...
+
+App.propTypes = {
+  sources: PropTypes.object,
+  selectSource: PropTypes.func,
+  selectedSource: PropTypes.object,
+  startPanelCollapsed: PropTypes.bool,
+  endPanelCollapsed: PropTypes.bool,
+};
+
+App.displayName = "App";
```
Comment (wldcordeiro) [inline src/components/App.js]: Could we use a getter for displayName and propTypes?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

File: src/components/App.js (hunk 2 above)
Comment (AnshulMalik) [inline src/components/App.js]: Hmm yes, but I was getting some flow errors if I don't define state out of constructor
`Did you forget to declare type parameter `State` of property `Component`?`

Any suggestions?

Comment (wldcordeiro) [inline src/components/App.js]: @AnshulMalik I see, didn't know Flow was the reason for it. I think we could make the first one a type annotation for Flow like what is discussed [here](https://github.com/facebook/flow/issues/1594).

File: src/components/App.js (hunk 3 above)
Comment (AnshulMalik) [inline src/components/App.js]: A quick question, making it getter will mean we are binding it to the object but I think propTypes should be attached to the class?

Comment (wldcordeiro) [inline src/components/App.js]: I think you would make it a static getter to attach to the class. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static

Comment (AnshulMalik) [inline src/components/App.js]: Another issue with static getter for propTypes
https://github.com/facebook/flow/issues/3008

Comment (jasonLaster) [inline src/components/App.js]: Yep - i think this is fine. We do this in another place as well and it seems like the current practice

File: src/main.js (hunk 1 above)
Comment (jasonLaster) [inline src/main.js]: @AnshulMalik this is good. I agree we should be consistent in a file.

**Pull request merged**